src/azure-cli/azure/cli/command_modules/profile/_help.py

```diff
@@ -36,11 +36,11 @@
     - name: Log in interactively.
       text: az login
     - name: Log in with username and password. This doesn't work with Microsoft accounts or accounts that have two-factor authentication enabled. Use -p=secret if the first character of the password is '-'.
-      text: az login -u johndoe@contoso.com -p VerySecret
-    - name: Log in with a service principal using client secret. Use -p=secret if the first character of the password is '-'.
-      text: az login --service-principal -u http://azure-cli-2016-08-05-14-31-15 -p VerySecret --tenant contoso.onmicrosoft.com
+      text: az login --username johndoe@contoso.com --password VerySecret
+    - name: Log in with a service principal using client secret. Use --password=secret if the first character of the password is '-'.
+      text: az login --service-principal --username APP_ID --password CLIENT_SECRET --tenant TENANT_ID
```

Comment:
`TENANT_ID` is preferred. Otherwise, a domain name will trigger the extra tenant discovery process:

https://github.com/Azure/azure-cli/blob/2548ad627b6e9714093b8e8bb60d7ddf6738c752/src/azure-cli/azure/cli/command_modules/profile/_validators.py#L12-L31